### PR TITLE
Fixes #25866 - handle missing puppet dist on upgrade task

### PR DESCRIPTION
**Pull request by jlsherrill**

### CODE REVIEW

**Comment by jlsherrill** [comment]
To reproduce, create a puppet repo in the UI/API, run this in the rails console:
```
repo = Katello::Repository.puppet_type.first; install_dist = repo.backend_service(SmartProxy.pulp_master).backend_data['distributors'].find { |dist| dist['distributor_type_id'] == 'puppet_install_distributor' };              SmartProxy.pulp_master.pulp_api.resources.repository.delete_distributor(repo.pulp_id, install_dist['id'])
```

Try to run the script:
```
rake katello:upgrades:3.11:update_puppet_repos
```

File: lib/katello/tasks/upgrades/3.11/update_puppet_repos.rake [hunk 1]
```diff
@@ -5,10 +5,16 @@ namespace :katello do
...
-          SmartProxy.pulp_master.pulp_api.resources.repository.delete_distributor(repo.pulp_id, install_dist['id'])
-          ForemanTasks.sync_task(::Actions::Pulp::Repository::Refresh, repo)
+          begin
+            puts "Refreshing repository #{repo.label} (#{repo.id})"
+            install_dist = repo.backend_service(SmartProxy.pulp_master).backend_data['distributors'].find { |dist| dist['distributor_type_id'] == 'puppet_install_distributor' }
+            if install_dist
+              SmartProxy.pulp_master.pulp_api.resources.repository.delete_distributor(repo.pulp_id, install_dist['id'])
+            end
+            ForemanTasks.sync_task(::Actions::Pulp::Repository::Refresh, repo)
+          rescue RestClient::ResourceNotFound
```
Comment (beav) [inline lib/katello/tasks/upgrades/3.11/update_puppet_repos.rake]: does the 404 ever happen as a result of normal-ish data?

Comment (jlsherrill) [reply lib/katello/tasks/upgrades/3.11/update_puppet_repos.rake]: it should not no, and if a repo is missing in pulp, it should get corrected by the correct_repos step/task anyways, but i figured it'd be better to handle it here as well

Comment (beav) [reply lib/katello/tasks/upgrades/3.11/update_puppet_repos.rake]: I would log a warning here that pulp_id was not found before continuing

**Comment by beav** [comment]
@jlsherrill I didnt test this, but the code change is very straightforward. ack pending the one comment I made to add a log statement

**Comment by jlsherrill** [comment]
@beav updated

**Pull request merged**